constraints.pro

```diff
@@ -0,0 +1,152 @@
+%%% Utility predicates
+
+is_valid_version_range(VersionRange) :-
+  VersionRange = 'workspace:^';
+  VersionRange = 'workspace:~';
+  parse_version_range(VersionRange, _, _, _, _).
+
+atom_to_number(Atom, Number) :-
+  atom_chars(Atom, Chars),
+  number_chars(Number, Chars).
+
+is_atom_number(Atom) :-
+  catch(atom_to_number(Atom, _), _, false).
+
+parse_version_range(VersionRange, Modifier, Major, Minor, Patch) :-
```

Comment:
Heeyyyy loook it's a bunch of Prolog 😆 

But seriously, let me know if any of this makes sense — if not, I'm happy to make it more readable.

Reply:
Maybe add one or two lines at the very top of this file with a link to https://yarnpkg.com/features/constraints/?

Not sure how widespread familiarity is with yarn constraints in general.

Reply:
Oops I didn't see this comment until now. I'll make a note to do this in a future PR.